### PR TITLE
[Eager] Polish grad code details

### PR DESCRIPTION
**Pull request by chenwhql**
### PR types

Function optimization
### PR changes

Others
### Describe

[Eager] Polish grad code details

阅读代码时，顺手完善了一些代码细节问题:

1. 一些细节代码规范问题
2. 一些类成员命名规范问题，结尾要带下划线
3. 部分函数适合放到.cc或者作为私有成员函数
4. 去掉RecordEvent的字符串拼接
5. 去掉不必要的map查找
6. 部分注释整理

### CODE REVIEW

**Review by veyron95 — DISMISSED** [review]
LGTM

File: paddle/fluid/eager/backward.cc [hunk 1]
```diff
@@ -164,9 +165,9 @@ class GeneralGrad {
       visited.insert(node);
 
       // Check node is target_nodes or not, if node is not target_node,
-      // all the next_node will be marked in potential_stop_nodes
-      bool is_potential_stop_nodes =
-          input_target_nodes_inputmeta_map.count(node);
+      // all the next_node will be marked in potential_stop_nodes_
+      bool is_potential_stop_nodes_ =
```
Comment (JiabinYang) [inline paddle/fluid/eager/backward.cc]: no need to be `_`?

File: paddle/fluid/eager/tensor_wrapper.h [hunk 2]
```diff
@@ -70,24 +70,25 @@ class TensorWrapper {
...
-    // shallow copy tensor_impl here
-    if (no_need_buffer) {
-      if (phi::DenseTensor::classof(tensor.impl().get())) {
-        // Only Copy Meta
-        phi::DenseTensor* dense_tensor =
-            static_cast<phi::DenseTensor*>(tensor.impl().get());
-        auto tw_dense_tensor = std::make_shared<phi::DenseTensor>();
-        tw_dense_tensor->set_meta(dense_tensor->meta());
-        intermidiate_tensor_.set_impl(tw_dense_tensor);
+    } else {
```
Comment (JiabinYang) [inline paddle/fluid/eager/tensor_wrapper.h]: since first part of if will return, it seems no need to build a extra else branch

**Review by JiabinYang — COMMENTED** [review]
some comments

File: paddle/fluid/eager/backward.cc (hunk 1 above)
Comment (chenwhql) [reply paddle/fluid/eager/backward.cc]: done, remove this suffix, thx

File: paddle/fluid/eager/tensor_wrapper.h (hunk 2 above)
Comment (chenwhql) [reply paddle/fluid/eager/tensor_wrapper.h]: done, revert this branch, thx

**Review by JiabinYang — APPROVED** [review]
LGTM

**Review by zhiqiu — APPROVED** [review]
LGTM

**Review by luotao1 — APPROVED** [review]

**Pull request merged**